### ISSUE TITLE
🐛 Bug Report: Authenticate API requests tutorial does not work

### ISSUE THREAD

**Issue by eipc16** [backstage/backstage#21824]
### 📜 Description

Authentication API Request tutorial does not work / has errors. Tutorial in question:  https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md

The error was most likely introduced in this MR: https://github.com/backstage/backstage/pull/14095.
This MR updated the tutorial to use `IdentityApi.getIdentity` method instead of the deprecated `IdentityApi.authenticate` method which operates on a provided token, but the `DefaultIdentityClient` only extracts the token from request header: https://github.com/backstage/backstage/blob/master/plugins/auth-node/src/identity/DefaultIdentityClient.ts#L87

Because of this, the token extracted from `token` cookie is passed to the `ServerTokenManager` instead of being used to resolve user Identity. This throws and error pasted below.

### 👍 Expected behavior

Cookie based authentication works (ex. in techdocs)

### 👎 Actual Behavior with Screenshots

An error is thrown when using cookies for auth (ex. techdocs).

Error:
```
[1] AuthenticationError: Invalid server token; caused by TypeError: Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer
[1]     at ServerTokenManager.authenticate (/project-dir/node_modules/@backstage/backend-common/dist/index.cjs.js:4121:11)
[1]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[1]     at <anonymous> (/project-dir/packages/backend/src/middlewares/auth/auth.ts:56:9) {
[1]   cause: TypeError: Key for the ES256 algorithm must be one of type KeyObject or CryptoKey. Received an instance of Buffer
[1]       at asymmetricTypeCheck (/project-dir/node_modules/jose/dist/node/cjs/lib/check_key_type.js:17:15)
[1]       at checkKeyType (/project-dir/node_modules/jose/dist/node/cjs/lib/check_key_type.js:44:9)
[1]       at flattenedVerify (/project-dir/node_modules/jose/dist/node/cjs/jws/flattened/verify.js:79:29)
[1]       at compactVerify (/project-dir/node_modules/jose/dist/node/cjs/jws/compact/verify.js:18:44)
[1]       at Object.jwtVerify (/project-dir/node_modules/jose/dist/node/cjs/jwt/verify.js:9:44)
[1]       at ServerTokenManager.authenticate (/project-dir/node_modules/@backstage/backend-common/dist/index.cjs.js:4104:24)
[1]       at <anonymous> (/project-dir/packages/backend/src/middlewares/auth/auth.ts:56:28)
[1]       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### 👟 Reproduction steps

1. Follow the tutorial: https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md
2. Trigget a request that does not contain Authorization header, but includes token cookie

### 🖥️ Your Environment

```
OS:   Linux 6.2.0-37-generic - linux/x64
node: v18.16.0
yarn: 1.22.1
cli:  0.23.1 (installed)
backstage:  1.19.6

Dependencies:
  @backstage/app-defaults                                          1.4.4
  @backstage/backend-app-api                                       0.5.8
  @backstage/backend-common                                        0.19.8, 0.19.9
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.0.5
  @backstage/backend-plugin-api                                    0.6.6, 0.6.7
  @backstage/backend-tasks                                         0.5.11, 0.5.12
  @backstage/backend-test-utils                                    0.2.7
  @backstage/catalog-client                                        1.4.5, 1.4.6
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.1.5, 0.2.0
  @backstage/cli                                                   0.23.1
  @backstage/config-loader                                         1.5.3
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.0
  @backstage/core-components                                       0.13.7
  @backstage/core-plugin-api                                       1.7.0
  @backstage/dev-utils                                             1.0.22
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.3
  @backstage/frontend-app-api                                      0.2.0
  @backstage/frontend-plugin-api                                   0.2.0
  @backstage/integration-aws-node                                  0.1.8
  @backstage/integration-react                                     1.1.20
  @backstage/integration                                           1.7.1, 1.7.2
  @backstage/plugin-api-docs                                       0.9.12
  @backstage/plugin-app-backend                                    0.3.54
  @backstage/plugin-app-node                                       0.1.6
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.0
  @backstage/plugin-auth-backend-module-github-provider            0.1.3
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.3
  @backstage/plugin-auth-backend-module-google-provider            0.1.3
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.3
  @backstage/plugin-auth-backend                                   0.19.4
  @backstage/plugin-auth-node                                      0.4.0, 0.4.1
  @backstage/plugin-bazaar-backend                                 0.3.4
  @backstage/plugin-bazaar                                         0.2.17
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.3
  @backstage/plugin-catalog-backend                                1.14.0
  @backstage/plugin-catalog-common                                 1.0.17
  @backstage/plugin-catalog-graph                                  0.2.37
  @backstage/plugin-catalog-node                                   1.4.7
  @backstage/plugin-catalog-react                                  1.8.5
  @backstage/plugin-catalog                                        1.14.0
  @backstage/plugin-events-node                                    0.2.15
  @backstage/plugin-graphiql                                       0.2.55
  @backstage/plugin-home-react                                     0.1.4
  @backstage/plugin-home                                           0.5.9
  @backstage/plugin-org                                            0.6.15
  @backstage/plugin-permission-backend                             0.5.30
  @backstage/plugin-permission-common                              0.7.10, 0.7.9
  @backstage/plugin-permission-node                                0.7.17, 0.7.18
  @backstage/plugin-permission-react                               0.4.16
  @backstage/plugin-proxy-backend                                  0.4.4
  @backstage/plugin-scaffolder-backend                             1.18.1
  @backstage/plugin-scaffolder-common                              1.4.2
  @backstage/plugin-scaffolder-node                                0.2.7
  @backstage/plugin-scaffolder-react                               1.5.6
  @backstage/plugin-scaffolder                                     1.15.1
  @backstage/plugin-search-backend-module-catalog                  0.1.10
  @backstage/plugin-search-backend-module-pg                       0.5.15
  @backstage/plugin-search-backend-module-techdocs                 0.1.10
  @backstage/plugin-search-backend-node                            1.2.10
  @backstage/plugin-search-backend                                 1.4.6
  @backstage/plugin-search-common                                  1.2.7
  @backstage/plugin-search-react                                   1.7.1
  @backstage/plugin-search                                         1.4.1
  @backstage/plugin-techdocs-backend                               1.8.0
  @backstage/plugin-techdocs-node                                  1.9.0
  @backstage/plugin-techdocs-react                                 1.1.12
  @backstage/plugin-techdocs                                       1.8.0
  @backstage/plugin-user-settings                                  0.7.11
  @backstage/release-manifests                                     0.0.10
  @backstage/test-utils                                            1.4.4
  @backstage/theme                                                 0.4.3
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.6
Done in 0.80s.
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by Josh-Uvi** [comment]
Hey @eipc16, I'll investigate this issue

**Comment by Josh-Uvi** [comment]
Hi @eipc16,

I followed the tutorial and the reproduction steps you provided, but I was unable to reproduce the issue. I tried using two different versions of Backstage, v1.18.1 and the current stable version, but the problem didn't occur in either case. 

I believe that the issue may have been resolved in the latest version, and I couldn't test it with your version because Backstage doesn't yet support creating new instances tied to a specific version. 

In conclusion, I suggest that you upgrade to a more recent version to avoid similar issues in the future.

cc @tudi2d

**Comment by chipnesh** [comment]
I encountered the same problem. However, it turned out that I had applied middleware to the auth endpoint, which was unnecessary.

When I removed the authMiddleware from the /auth endpoint, everything started working perfectly.

In `packages/backend/src/index.ts`

Before:
```
apiRouter.use('/auth', authMiddleware, await auth(authEnv));
```

After:
```
apiRouter.use('/auth', await auth(authEnv));
```

**Comment by Rugvip** [comment]
Will close this since we now strongly recommend against using this tutorial. New Backstage projects have this protection built-in by default, and old projects can [be migrated](https://backstage.io/docs/tutorials/auth-service-migration).
